quic/s2n-quic-platform/src/syscall.rs

```diff
@@ -60,6 +63,23 @@ pub fn bind_udp<A: std::net::ToSocketAddrs>(addr: A, reuse_port: bool) -> io::Re
     Ok(socket)
 }
 
+/// Binds a socket to a specified interface by name
+#[cfg(target_os = "linux")]
+#[allow(dead_code)] // This is currently only used in the XDP io provider, which is optional
```

Comment:
Would this make sense for other IO providers to use, or could it be `#cfg(all(target_os = "linux", feature = "xdp"))`?

Reply:
Yeah I'll just gate it with `feature = "xdp"`